### ISSUE TITLE
knex oracle dialect sets UV_THREADPOOL_SIZE without knowing the effect

### ISSUE THREAD

**Issue by tsondergaard** [knex/knex#3715]
lib/dialects/oracledb/index.js has this code:

```
  // Node.js only have 4 background threads by default, oracledb needs one by connection
  if (this.driver) {
    process.env.UV_THREADPOOL_SIZE = process.env.UV_THREADPOOL_SIZE || 1;
    process.env.UV_THREADPOOL_SIZE =
      parseInt(process.env.UV_THREADPOOL_SIZE) + this.driver.poolMax;
  }
```
If the libuv threadpool has been initialized before this code is executed, this does not have the desired effect as UV_THREADPOOL_SIZE is only read once, when libuv initializes the threadpool. I don't think it is correct for the knex oracle dialect to do this. First of all because changing UV_THREADPOOL_SIZE at an arbitrary time during runtime is unreliable, but also because it is something for the actual oracle driver to work out.

I think the knex oracle driver should limit itself to logging a debug message if `(UV_THREADPOOL_SIZE || 4) < this.driver.poolMax`.

**Comment by briandamaged** [comment]
I agree that this is something that should be investigated further.  I'm also concerned about how this code will behave in multi-client / multi-pool scenarios.  (Speaking of which... the `(UV_THREADPOOL_SIZE || 4) < this.driver.poolMax` check probably would not be sufficient in this scenario.  The check would work for each client individually, but not collectively)

**Comment by mercmobily** [comment]
You’re right to be wary of libraries changing process‑wide settings.

The oracledb dialect bumps `process.env.UV_THREADPOOL_SIZE` by `poolMax` when the driver is present (lib/dialects/oracledb/index.js:24–40). The intent was to make sure libuv has enough threads to service Oracle I/O, but doing this implicitly can have unintended effects for applications managing their own thread‑pool tuning.

Maybe we should stop  mutating `UV_THREADPOOL_SIZE` silently; instead, document a recommended size (e.g. ≥ driver.poolMax + baseline) and log a warning if the current value is obviously too small. If we keep any mutation, gate it behind an explicit config flag.
